### PR TITLE
feat: Add `--type` flag to `kpt fn eval`

### PR DESCRIPTION
**Pull request by yuwenma**
Add new flag `--type` to `kpt fn eval.  This flag works together with `--save` which will add the evaluated functions to the right Kptfile pipeline section ("validators" or "mutators")

Misc: 	
- Add e2e test `save-fn/validator-type`
- Update README and autogenerate the user guides
- Somehow, the doc for a previous flag `--save` is droppped. https://github.com/GoogleContainerTools/kpt/commit/50230e355398b7e472ffe1de7854a1eb84e08a62 Added the doc for `--save` as well.

### CODE REVIEW

File: site/reference/cli/fn/eval/README.md [hunk 1]
```diff
@@ -125,11 +125,21 @@ fn-args:
...
+
 --results-dir:
   Path to a directory to write structured results. Directory will be created if
   it doesn't exist. Structured results emitted by the functions are aggregated and saved
   to `results.yaml` file in the specified directory.
   If not specified, no result files are written to the local filesystem.
+  
+--save, s:
+  Save the function image and fn-config to Kptfile. Require ` + "`" + `--image` + "`" + `.
+
```
Comment (droot) [inline site/reference/cli/fn/eval/README.md:142]: You can separate the docs change in a separate PR and merge them after the kpt release.

This is because docs change will go live on the site immediately once the PR is merged but kpt's latest release won't have the changes.

Comment (yuwenma) [reply site/reference/cli/fn/eval/README.md:142]: oh! Now I recalled why we removed the --save docs. kk.

Comment (yuwenma) [reply site/reference/cli/fn/eval/README.md]: Separate the docs in another PR https://github.com/GoogleContainerTools/kpt/pull/2977
